vcap_utils/vcap_utils/backends/base_openvino.py

```diff
@@ -35,34 +35,112 @@ def __init__(self, model_xml: bytes,
+        # If the device is a MULTI device, use the default num_requests
+        is_multi_device = device_name.lower().startswith("multi")
+
+        num_requests = 0
+        if not is_multi_device:
+            # Find the optimal number of InferRequests for this device
+            supported_metrics = self.ie.get_metric(
+                device_name, _SUPPORTED_METRICS)
+            if _RANGE_FOR_ASYNC_INFER_REQUESTS in supported_metrics:
+                low, high, _ = self.ie.get_metric(
+                    device_name, _RANGE_FOR_ASYNC_INFER_REQUESTS)
+                # Cap the num_requests, because setting it too high can crash
+                # TODO(Alex): Figure out _why_ hddl crashes when set to 'high'
+                num_requests = max(0, min(low * 2, high))
 
         self.net: IENetwork = self.ie.read_network(
             model=model_xml,
             weights=weights_bin,
             init_from_buffer=True)
 
-        self.exec_net: ExecutableNetwork = self.ie.load_network(
-            network=self.net,
-            device_name=device_name,
-            num_requests=n_requests)
+        try:
+            self.exec_net: ExecutableNetwork = self.ie.load_network(
+                network=self.net,
+                device_name=device_name,
+                num_requests=num_requests)
+        except RuntimeError as e:
+            if device_name == "CPU":
+                # It's unknown why this would happen when loading
+                # onto CPU, so we re-raise the error.
+                raise e
+
+            # This error happens when trying to load onto HDDL or MYRIAD, but
+            # somehow the device fails to work. In these cases, we try again
+            # but load onto 'CPU', which is a safe choice.
+            msg = f"Failed to load {self.__class__} onto device " \
+                  f"{device_name}. Error: '{e}'. " \
+                  f"Trying again with device_name='CPU'"
+            logging.critical(msg)
+            self.__init__(
+                model_xml=model_xml,
+                weights_bin=weights_bin,
+                device_name="CPU",
+                ie_core=ie_core)
+            return
 
         # Pull out a couple useful constants
-        self.InferRequestStatusCode = StatusCode
         self.input_blob_names: List[str] = list(self.net.inputs.keys())
         self.output_blob_names: List[str] = list(self.net.outputs.keys())
 
+        # For running threaded requests to the network
+        self._get_free_request_lock = RLock()
+        self._request_free_events: List[Event] = [
+            Event() for _ in self.exec_net.requests]
+        for request_free in self._request_free_events:
+            request_free.set()
+
+        # For keeping track of the workload for this backend
+        self._total_requests: int = len(self.exec_net.requests)
+        self._num_ongoing_requests: int = 0
+        self._num_ongoing_requests_lock: RLock = RLock()
+
+    @property
+    def workload(self) -> float:
```

Comment:
A custom 'workload' measurement for the BaseOpenVINOBackend makes sense, since the exec_net has a certain num_requests to it- so the 'workload' measurement becomes 'num ongoing requests' / 'total possible ongoing requests'

Reply:
This won't affect much unless someone makes a custom DeviceMapper filter that allows multiple Backends to be loaded, eg, a backend for CPU and a backend for HDDL. In those cases, this workload measurement will be used heavily to figure out which backend is busier.

Reply:
Should this (and some of the other comments you left in the review) be added as comments in the code itself?

Reply:
Well, the comment on the function tells you what the function does... I could augment it to explain my last point (About DeviceMappers)

Reply:
FYI- those comments were my attempt to make this MR less of a "going in dry" and more of a tour guide, to help y'all grok through why things changed, rather than just what changed

Reply:
Future code readers will want the guided tour as well, without having to pull up this merge request

Reply:
I don't think that's how it works though. I'm explaining the things I changed. It doesn't translate 1:1 to what might go into docstrings...

I was trying to improve the MR experience. If you have improvements on docs, specify which docstrings could be improved, and how